[standards] Rewrite imports in RNTester to use standard paths

ide opened the pull request:
## Summary

See https://github.com/facebook/react-native/issues/24316 for the motivation. This commit rewrites the imports in the RNTester project.

## Changelog

[General] [Changed] - Replaced Haste-style imports with standard path-style imports for RNTester

## Test Plan

Run RNTester and verify that it loads without any issues. Tap through each screen to ensure the JS modules load (that is, were bundled) correctly.

cpojer commented:
Instead of the relative paths, what do you think about doing `require('react-native/Libraries/…')`?

ide commented:
- Updated requires to use the public interface where possible (`const {View} = require('react-native')`)
- Changed `const ReactNative = require('react-native'); const {...} = ReactNative;` to `const {...} = require('react-native');`, which sets us up to move to import/export more easily
- I left requires of internal modules (utilities, for example) alone, since React Native may want to seal them off entirely outside of the Facebook repo and not encourage `require('react-native/Libraries/Utilities/...')`. React in particular publishes itself pre-bundled to intentionally separate its public interface from its internals and prevent `require('react/...')`.
